examples/shors_code.py

```diff
@@ -0,0 +1,118 @@
+""" Shor's code is a stabilizer code for quantum error correction.
+It uses 9 qubits to encode 1 logic qubit and is resistant
+to at least one bit flip and one sign flip or their combination.
+
+(0, 0): ───@───@───H───@───@───@───@───X───H───@───@───X───M───
+           │   │       │   │   │   │   │       │   │   │
+(0, 1): ───┼───┼───────X───┼───X───┼───@───────┼───┼───┼───M───
+           │   │           │       │   │       │   │   │
+(0, 2): ───┼───┼───────────X───────X───@───────┼───┼───┼───M───
+           │   │                               │   │   │
+(0, 3): ───X───┼───H───@───@───@───@───X───H───X───┼───@───M───
+               │       │   │   │   │   │           │   │
+(0, 4): ───────┼───────X───┼───X───┼───@───────────┼───┼───M───
+               │           │       │   │           │   │
+(0, 5): ───────┼───────────X───────X───@───────────┼───┼───M───
+               │                                   │   │
+(0, 6): ───────X───H───@───@───@───@───X───H───────X───@───M───
+                       │   │   │   │   │
+(0, 7): ───────────────X───┼───X───┼───@───────────────────M───
+                           │       │   │
+(0, 8): ───────────────────X───────X───@───────────────────M───
+"""
+
+import random
+
+import cirq
+
+
+class OneQubitShorsCode:
+
+    def __init__(self):
+        self.num_physical_qubits = 9
+        self.physical_qubits = cirq.LineQubit.range(self.num_physical_qubits)
+
+    def encode(self):
+        yield cirq.ops.Moment(
+            [cirq.CNOT(self.physical_qubits[0], self.physical_qubits[3])])
+        yield cirq.ops.Moment(
+            [cirq.CNOT(self.physical_qubits[0], self.physical_qubits[6])])
+        yield cirq.ops.Moment([
+            cirq.H(self.physical_qubits[0]),
+            cirq.H(self.physical_qubits[3]),
+            cirq.H(self.physical_qubits[6])
+        ])
+        yield cirq.ops.Moment([
+            cirq.CNOT(self.physical_qubits[0], self.physical_qubits[1]),
+            cirq.CNOT(self.physical_qubits[3], self.physical_qubits[4]),
+            cirq.CNOT(self.physical_qubits[6], self.physical_qubits[7])
+        ])
+        yield cirq.ops.Moment([
+            cirq.CNOT(self.physical_qubits[0], self.physical_qubits[2]),
+            cirq.CNOT(self.physical_qubits[3], self.physical_qubits[5]),
+            cirq.CNOT(self.physical_qubits[6], self.physical_qubits[8])
+        ])
+
+    def apply_gate(self, gate: cirq.Gate, pos: int):
+        if pos > self.num_physical_qubits:
+            raise IndexError
+        else:
+            return gate(self.physical_qubits[pos])
+
+    def correct(self):
+        yield cirq.ops.Moment([
```

Comment:
This is not correct. 

To correct errors in an error correction code, you have to first measure the syndromes, and use those results. I'm really not sure what this code does - it looks like it reverses the encoding - but I don't think that's how you decode per se - I think you'll have to measure the right Zlogical =XXXXXXXX and Xlogical=ZZZZZZZZZ observables, but I really want to find a good description to prove my hunch here.

Reply:
For Shor's code, the "correct" step is the same as long as the errors is within the scope of the code. No need to measure before correct. 
Basically, you do the encode and the error (X, Z or Y) happens or no error, and then performs correct. Then you measure the first qubit in Z basis, the result is corrected original qubit.

Usually stablizer code needs to measure before correct because the correct will depend on the result of the measurement. But Cirq doesn't have classical register, so we are not able to implement such codes for the time being.

Reply:
I'm standing corrected (no pun intended). After discussions with the team and you, we can skip the stabilizer formalism aspect of the Shor code. We can focus on the encode and decode/correct form.

Regarding the classical register - that is true, however, in Cirq you can create conditional gates on measured qubits, which would work similarly well, as seen in our [quantum_teleportation.py](https://github.com/quantumlib/Cirq/blob/master/examples/quantum_teleportation.py) example